src/Workspaces/Core/Portable/PublicAPI.Unshipped.txt

```diff
@@ -1,5 +1,14 @@
+abstract Microsoft.CodeAnalysis.CodeFixes.FixAllContext.SpanBasedDiagnosticProvider.GetDocumentSpanDiagnosticsAsync(Microsoft.CodeAnalysis.Document document, Microsoft.CodeAnalysis.Text.TextSpan fixAllSpan, System.Threading.CancellationToken cancellationToken) -> System.Threading.Tasks.Task<System.Collections.Generic.IEnumerable<Microsoft.CodeAnalysis.Diagnostic>>
+Microsoft.CodeAnalysis.CodeFixes.DocumentBasedFixAllProvider.DocumentBasedFixAllProvider(System.Collections.Immutable.ImmutableArray<Microsoft.CodeAnalysis.CodeFixes.FixAllScope> supportedFixAllScopes) -> void
```

Comment:
Are these public API additions intentional?

Reply:
I think we can duplicate some types here and make them internal.

Reply:
Yes, there's an issue template for it. Manish, you're one of the area owners so go ahead and file the issue and apply the `api-ready-for-review` tag.

Reply:
@jmarolf Yes, these APIs are intentionally public. Otherwise, public clients such as MS.CA.Testing library, would be unable to create a FixAll provider that supports these new FixAll scopes. We can certainly keep them internal in this PR and have a separate PR to just make these APIs public.

Reply:
https://github.com/dotnet/roslyn/issues/60029 tracks making APIs public. I'll switch the APIs to internal for this PR

Reply:
I have internalized most of these APIs with https://github.com/dotnet/roslyn/pull/59989/commits/2df4e07f971dc80771b249de77e04c453eea0d9f. I still had to retain one public API. I will open an issue with the public API proposals.